src/contact.rs

```diff
@@ -379,6 +382,18 @@ impl Contact {
             bail!("Bad address supplied: {:?}", addr);
         }
 
+        let mut name = name.as_ref();
+        if origin <= Origin::OutgoingTo {
+            if addr.contains("noreply") || (addr.len() > 45 && addr.contains('+'))
+            // Filter out email addresses where the user accidentally wrote to
+            // and use-once addresses (like reply+AEJ...@reply.github.com)
+            {
+                info!(context, "hiding contact {}", addr);
+                origin = Origin::Hidden;
+                name = "";
+            }
+        }
```

Comment:
When testing mailing lists, I noticed that sometimes a mailing list
contact got a name (like, hocuri <noreply@github.com>). It turned out
that ages ago, I had accidentally written an email to - in this example
- hocuri <noreply@github.com> and it had been added to the contacts
list.

This hides email addresses from the contacts list that are obviously not
meant to be written at and prevents updating the names.